### PR TITLE
Consolidate events between EventConstants and BrowserEventEmitter

### PR DESCRIPTION
**Pull request by nhunzaker**
This is a follow up to the request made by @spicyj in https://github.com/facebook/react/pull/9333. This commit removes some duplication of event names by consolidating them into the EventConstants module. It then renames that module to BrowserEventConstants.

### CODE REVIEW

File: docs/js/react-dom.js [hunk 1]
```diff
@@ -350,7 +350,7 @@ function extractCompositionEvent(topLevelType, targetInst, nativeEvent, nativeEv
 }
 
 /**
- * @param {string} topLevelType Record from `EventConstants`.
```
Comment (spicyj) [inline docs/js/react-dom.js]: can you revert this file?

File: src/renderers/shared/shared/event/BrowserEventConstants.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule BrowserEventConstants
+ */
+
+'use strict';
+
+var getVendorPrefixedEventName = require('getVendorPrefixedEventName');
+
+export type PropagationPhases = 'bubbled' | 'captured';
```
Comment (spicyj) [inline src/renderers/shared/shared/event/BrowserEventConstants.js]: unnecessary?

Comment (nhunzaker) [inline src/renderers/shared/shared/event/BrowserEventConstants.js]: Rollover from [EventConstants]https://github.com/facebook/react/blob/master/src/renderers/shared/shared/event/EventConstants.js#L14). Happy to remove though.

File: docs/js/react-dom.js (hunk 1 above)
Comment (nhunzaker) [inline docs/js/react-dom.js]: Sure thing.

File: src/renderers/shared/shared/event/BrowserEventConstants.js (hunk 2 above)
Comment (nhunzaker) [inline src/renderers/shared/shared/event/BrowserEventConstants.js]: Ah right, it gets referenced here: https://github.com/facebook/react/pull/9512/files#diff-1ca53ffa4da75a64f7c3b2f131f62da7R21

File: docs/js/react-dom.js (hunk 1 above)
Comment (nhunzaker) [inline docs/js/react-dom.js]: Done!

File: src/renderers/shared/shared/event/BrowserEventConstants.js (hunk 2 above)
Comment (spicyj) [inline src/renderers/shared/shared/event/BrowserEventConstants.js]: let's just move it to that file

Comment (nhunzaker) [inline src/renderers/shared/shared/event/BrowserEventConstants.js]: Done in caccf43

**Comment by nhunzaker** [comment]
I think everything should be good to go. Struggling with CI however. Any ideas?

**Comment by spicyj** [comment]
```
'getVendorPrefixedEventName' is imported by src/renderers/shared/shared/event/BrowserEventConstants.js, but could not be resolved – treating it as an external dependency
```

Does moving the file to src/renderers/dom/shared/event/BrowserEventConstants.js fix the issue? That is where it should be anyway.

**Comment by nhunzaker** [comment]
@spicyj Hey, that did it! I just upstreamed from master, but looks like that was it!

https://circleci.com/gh/facebook/react/2923

**Comment by spicyj** [comment]
Thanks!

**Pull request merged**